lambdas/lambda_function.py

```diff
@@ -104,10 +112,26 @@ def key_rotate(appsync_client, env):
                         ]
                     },
                 },
-                {"name": "targetContainers", "data": secrets["BAW_CONTAINERS"],},
+                {
+                    "name": "targetContainers",
+                    "data": secrets["BAW_CONTAINERS"],
+                },
                 {"name": "defaultOnly", "data": env["default_only"]},
             ]
         },
     )
     if r.status_code != 201:
         raise Exception(r.text)
+
+
+def _delete_expired_keys(appsync_client, env):
+    """Delete keys that have already expired, to avoid exceeding the limit on the number of API keys"""
+    paginator = appsync_client.get_paginator("list_api_keys")
+    response_iterator = paginator.paginate(apiId=env["apiId"])
+
+    now = time.time()
+
+    for response in response_iterator:
+        for key in response["apiKeys"]:
+            if key["expires"] < now:
+                appsync_client.delete_api_key(apiId=env["apiId"], id=key["id"])
```

Comment:
do we know how many keys are valid at any particular time at the moment?

Reply:
7